File: demo/src/pages/AnimationDemo/AnimationDemo.ts
import {main, Button, State, Text, Font, degrees, Viewable, Bound, Color, Double} from "@tswift/ui";

@main
export class AnimationDemo extends Viewable {

    @State angle:Double = 0;
    @State color:Color = Color.blue;
    @State isBold: Boolean = false;
    @State  opacity = 1.0

    
    body = ({ opacity, $opacity, color, isBold, angle, $angle }: Bound<this>) =>[
        Text("Hello, World")
        .padding()
        .foregroundColor(color)
        .font(Font.title.weight(isBold ? '.bold' : '.regular'))
        .rotationEffect(degrees(angle))
        .onTapGesture(() => {
            this.angle += 90;
            this.color = color == Color.blue ? Color.red : Color.blue;
            this.isBold = !this.isBold;
        })
        .animation('.easeInOut', $angle),
        Button("Press here", ()=>
            withAnimation(()=> $opacity(opacity -= 0.2));
        )
        .padding()
        .opacity(opacity)
    ];
        
}

File: demo/src/pages/AnimationDemo/AnimationDemo.swift
import SwiftUI

struct AnimationDemo: View {
    
    @State private var angle: Double = 0;
    @State private var color: Color = .blue;
    @State private var isBold: Bool = false;
    
    var body: some View {
        Text("Hello, World!")
            .padding()
            .foregroundColor(color)
            .font(.title.weight(isBold ? .bold : .regular))
            .rotationEffect(.degrees(angle))
            .onTapGesture {
                angle += 90
                color = (color == .blue ? .red : .blue);
                isBold.toggle()
                
            }.animation(.easeInOut, value:angle )
    }
}

struct AnimationDemo_Previews: PreviewProvider {
    
    static var previews: some View {
        AnimationDemo()
    }
}
